code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADIdentityProvider.java

```diff
@@ -277,4 +286,62 @@ protected AbstractTokenUtil getTokenUtils() {
     public String getName() {
         return getConstantsConfig().getProviderName();
     }
+
+    public List<Identity> savedIdentities() {
+        List<String> ids = fromHashSeparatedString(ADConstants.AD_ALLOWED_IDENTITIES.get());
+        List<Identity> identities = new ArrayList<>();
+        if (ids.isEmpty() || !isConfigured()) {
+            return identities;
+        }
+        for(String id: ids){
+            String[] split = id.split(":", 2);
+            identities.add(getIdentity(split[1], split[0]));
+        }
+        return identities;
+    }
+
+    public List<String> fromHashSeparatedString(String string) {
```

Comment:
Would be cleaner if we move all separator related logic to some util class? Applying # separator in one class, and assuming this separator in another class, is not quite obvious.